### DISCUSSION TITLE
Airfflow "Code View" not displaying the actual DAG code

### DISCUSSION THREAD

**Discussion by sarthakhaikerwalsaxena** [apache/airflow#24521] [General]
### Apache Airflow version

2.3.2 (latest released)

### What happened

We use DAG bags to parse DAGs across different custom directories. The Code View in the DAG rather than showing the actual DAG code, is displaying the code which is used to parse DAGs. 

![image](https://user-images.githubusercontent.com/11522831/174260493-48b9832f-7e5e-4503-b820-94822b1b8d9e.png)

### What you think should happen instead

It should display the actual DAG code. Note that its working fine in 2.1.2. Once we upgraded to 2.3.2 we are seeing this issue. 

### How to reproduce

You can replicate this issue by using DAG bags. 

### Operating System

RHEL 7.9

### Deployment

Virtualenv installation

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
I don't think using DagBag like this has been ever supported. Maybe it worked by accident in the past (though I am not sure how)  but I do not think it's even been "supported". IMHO DagBag is really an internal mechanism of Airflow, It can change it's behaviour at any time and the pattern you present is nowhere to be found in our documentation (just checked). 

I suggest you change to the "standard" method of dag storage - i.e. make sure that your DAGs are all available in the dags folder.

I will convert it into a discussion, and would love to hear what others think (@ashb @kaxil @dstandish ?) but I believe we are not going to do anything to change the behaviour ans support the way you invented.

**Reply by sarthakhaikerwalsaxena** [reply]
We have a requirement to use custom directories and sub-folders to use for different projects and teams. These directories also have different unix permissions enforced for multi-tenant security. 
We followed this blog for the initial implementation to use DAG bags to suit our requirement: https://xnuinside.medium.com/how-to-load-use-several-dag-folders-airflow-dagbags-b93e4ef4663c

We can change it to the standard storage mechanism under dags folder if thats the way it should be done. What do you suggest, should we do this or there is any way to get the code view working the way we are currently using it?

**Reply by potiuk** [reply]
The fact that someone wrote a blog post 4 years ago using Airflow version 1.10 which is end-of-life for more than a year already, does not mean the article is valid today. I have no opinion on that article, I do not really want to loose my time on reading something that is almost guaranteed to be out-dated.

 I'd strongly recommend you follow standard documentation of airflow - especially that you can choose the documentation version that is bound to the Airflow Version you are using.

My advice- whenever someone asks I always recommend people in similar situations to use `git-sync` and store their DAGs in Git - you can even use submodules, and separate repos if you want to keep separation between your teams. 

Airflow Summit 2022 just finished 3 weeks ago and you can find my summary of All the talks (all of the fresh and relevant to the current version of Airflow): you can learn a lot about good engineering practices (among other things)

https://potiuk.com/airflow-summit-2022-the-best-of-373bee2527fa

I highly recommend all the talks in `Make the best use of Airflow` but especially in your case `Engineering practices in your data pipelines` - especially the Jagex talk there might give you a lot of answers on how to make your data pipelines right.
